### ISSUE TITLE
Foreign key + search + pagination

### ISSUE THREAD

**Issue by ElForastero** [dfahlander/Dexie.js#1043]
First of all, thank you for your work. 🙏👍
Dexie is really like a breaf of fresh air for offline and client-side only apps.

I'm using Dexie v3 in a feeds reader app. The database consists of `sources` and `feeds` tables.

```ts
this.version(1).stores({
      sources: '++id',
      feed: '&id, [source_id+published], published, *titleTokens',
    });
```

There're 4 main cases for queries:

1. Query all `feeds` ordered by `published` date
2. Query all `feeds` where `*titleTokens` contains `search`, ordered by `published` date
3. Query all `feeds` for specific `source_id` ordered by `published` date
4. Query all `feeds` for `source_id` where `*titleTokens` contains `search`

Also, I need a working `pagination` and a `count` query in every case. 😁

I'm trying to find a best way to satisfy all conditions and not maintain 4 absolutely different queries.

The only solution that comes into my mind looks like this:

```ts
  let itemsCount;
  let identifiers;

  if (search === null && sourceID === null) {
    const query = db.feed.orderBy('published').reverse().offset(offset).limit(limit);
    identifiers = new Set(await query.primaryKeys());
    itemsCount = await db.feed.count();
  } else if (search === null && sourceID !== null) {
    const query = db.feed
      .where('[source_id+published]')
      .between([Number(sourceID), Dexie.minKey], [Number(sourceID), Dexie.maxKey])
      .offset(offset)
      .limit(limit);
    identifiers = new Set(await query.primaryKeys());
    itemsCount = await db.feed.where('source_id').equals(Number(sourceID)).count();
  } // and so on

  // Query feeds
  const feed = await db.feed.where('id').anyOf(identifiers).toArray();

  // JOIN sources
  await Promise.all(
    feed.map(async item => {
      item.source = await db.sources.get(item.source_id);
      item.source.__typename = 'Source';
    })
  );
```

Basically, I can make it work. But sometimes there will be many extra queries, especially for cases with `search`. Also I'm worrying about performance with, let's say, 100k `feeds` records.

Can you advice semething for my case? Is there a more elegant way to solve this?
